redisearch/autocomplete.go

```diff
@@ -60,27 +66,38 @@ func (a *Autocompleter) AddTerms(terms ...Suggestion) error {
 // Suggest gets completion suggestions from the Autocompleter dictionary to the given prefix.
 // If fuzzy is set, we also complete for prefixes that are in 1 Levenshten distance from the
 // given prefix
-func (a *Autocompleter) Suggest(prefix string, num int, fuzzy bool) ([]Suggestion, error) {
+func (a *Autocompleter) Suggest(prefix string, num int, fuzzy, withPayloads bool) ([]Suggestion, error) {
```

Comment:
This breaks the backwards compatibility of the API and we don't want that. Leave that function as is, and then add another one. Two options here:

1. SuggestWithPayload(...)

or

2. SuggestOpts(prefix string, SuggestOptions opt) {...}

where SuggestOptions is a struct with num, fuzzy, withPayloads and WITHSCORES (which is also not currently supported).

2 is more in line with the way we do things. We can have the old function return a warning that it's deprecated the first time it is called, and later remove it.

Reply:
I have chosen to follow option 2, the only thing I was unsure about was the depreciation message. I have included this in a comment in the godoc, but not sure if you mean something else. Let me know if what I have done is incorrect